Haploid mode

jydu opened the pull request:
I have introduced a few small changes in order to allow computation of statistics with haploid data. When enabling the --haploid switch, vcftools will assume that genotypes are haploid instead of diploid. This has two consequences: (1) an error will be sent in case a diploid genotype is found at some locus (without the switch, it currently complains in the presence of a haploid genotype) and (2) Ne parameters are divided by 2. This currently only affect the calculations of pi, Tajima's D and Fst values.

Commit:
Enable haploid mode

Added a --haploid switch to allow computation of pi and Fst and Tajima's D for haploid data.

Commit:
Fixed typos.

Commit:
Merge https://github.com/vcftools/vcftools

BioinfoNovice commented:
Hi I did install the latest version of vcftools (v01.0.15) and it does not seem to have this option --haploid.
Do I have to do any supplement manipulation ?

Yannovsky commented:
Hello,

Can I add the option --haploid to vcftools which was install with linuxbrew ?

if not how can we add the option to vcftools ?

jydu commented:
Hi,

The --haploid mode is an option I added to VCFtools, as a fork. As the proposed merge has not been accepted in the main repository, it is not part of the official release. If you want to use it, you have to compile VCFtools from source by cloning the forked repository, not the original one.

Hope this helps,

Julien.

cfyin commented:
Hi,
I ran this version with --haploid, the process not print the official version said "Fst: Only using diploid sites."
but the result file is empty?
Weir and Cockerham mean Fst estimate: nan
Weir and Cockerham weighted Fst estimate: nan
Why?

Commit:
Merge https://github.com/vcftools/vcftools

Commit:
Fixed nb of selected sites.

cfyin commented:
Thank you very much, jydu,
I downloaded your version at https://github.com/jydu/vcftools
But the result is just nan as the same as the previous one I downloaded from https://github.com/vcftools/vcftools/tree/3841041cbee32d0a0480ccabd4550b5502726d9b

jydu commented:
Hi,

Can you share minimal VCF files (for instance truncated) as well as the exact command line that you are using so that I give it a closer look?

cfyin commented:
I have email some lines to you.
Thank you !

jydu commented:
Sorry, did not get that. You'll find my email address here https://www.evolbio.mpg.de/person/40579/15929
(for later discussion without spamming the vcftools repository too much ;) )

cfyin commented:
##fileformat=VCFv4.2
...
##samtoolsVersion=0.1.19-44428cd
#CHROM	POS	ID	REF	ALT	QUAL	FILTER	INFO	FORMAT	variant	variant10	variant11	variant12	variant13	variant14	variant15	variant2	variant3	variant4	variant5	variant6	variant7	variant8	variant9
8	35518	.	G	C	753	.	AC=4;AF=1.00;AN=4;DP=147;MLEAC=1;MLEAF=1.00;MQ=60.00;MQ0=0;set=variant-variant2-variant6-variant13	GT:AD:DP:GQ:PL	1:0,32:32:99:783,0	.	.	.	1:0,28:28:99:704,0	.	.	1:1,34:35:99:840,0	.	.	.	1:2,50:52:99:1245,0	.	.	.
8	35625	.	C	T	720	.	AC=4;AF=1.00;AN=4;DP=158;MLEAC=1;MLEAF=1.00;MQ0=0;set=variant-variant2-variant6-variant13	GT:AD:DP:GQ:PL	1:3,36:39:99:750,0	.	.	.	1:0,26:26:99:599,0	.	.	1:0,47:47:99:1113,0	.	.	.	1:2,44:46:99:1041,0	.	.	.
9	64	.	G	C	6596	.	AC=2;AF=1.00;AN=2;DP=591;MLEAC=1;MLEAF=1.00;MQ=60.00;MQ0=0;set=variant6-variant7	GT:AD:DP:GQ:PL	.	.	.	.	.	.	.	.	.	.	.	1:10,280:290:99:6626,0	1:2,292:294:99:6946,0	.	.
9	85	.	A	T	2143	.	AC=2;AF=1.00;AN=2;DP=115;FS=0.000;MLEAC=1;MLEAF=1.00;MQ=60.00;MQ0=0;set=variant10-variant14	GT:AD:DP:GQ:PL	.	1:0,55:55:99:2173,0	.	.	.	1:0,60:60:99:2318,0	.	.	.	.	.	.	.	.	.
9	203	.	T	G	2615	.	AC=1;AF=1.00;AN=1;DP=57;FS=0.000;MLEAC=1;MLEAF=1.00;MQ=60.00;MQ0=0;QD=27.35;SOR=0.960;set=variant14	GT:AD:DP:GQ:PL	.	.	.	.	.	1:0,57:57:99:2645,0	.	.	.	.	.	.	.	.	.
9	2905	.	A	C	2618	.	AC=1;AF=1.00;AN=1;DP=57;FS=0.000;MLEAC=1;MLEAF=1.00;MQ=60.00;MQ0=0;QD=27.60;SOR=0.877;set=variant14	GT:AD:DP:GQ:PL	.	.	.	.	.	1:0,57:57:99:2648,0	.	.	.	.	.	.	.	.	.
9	2906	.	G	T	2611	.	AC=1;AF=1.00;AN=1;DP=57;FS=0.000;MLEAC=1;MLEAF=1.00;MQ=60.00;MQ0=0;QD=35.09;SOR=0.877;set=variant14	GT:AD:DP:GQ:PL	.	.	.	.	.	1:0,57:57:99:2641,0	.	.	.	.	.	.	.	.	.
9	2913	.	A	G	2608	.	AC=1;AF=1.00;AN=1;DP=58;FS=0.000;MLEAC=1;MLEAF=1.00;MQ=60.00;MQ0=0;QD=33.34;SOR=0.836;set=variant14	GT:AD:DP:GQ:PL	.	.	.	.	.	1:0,58:58:99:2638,0	.	.	.	.	.	.	.	.	.
9	2704	.	G	T	2491	.	AC=2;AF=1.00;AN=2;DP=128;FS=0.000;MLEAC=1;MLEAF=1.00;MQ=60.00;MQ0=0;set=variant10-variant14	GT:AD:DP:GQ:PL	.	1:0,62:62:99:2521,0	.	.	.	1:0,66:66:99:2681,0	.	.	.	.	.	.	.	.	.
9	2773	.	T	C	2627	.	AC=2;AF=1.00;AN=2;DP=133;FS=0.000;MLEAC=1;MLEAF=1.00;MQ=60.00;MQ0=0;set=variant10-variant14	GT:AD:DP:GQ:PL	.	1:0,67:67:99:2657,0	.	.	.	1:0,66:66:99:2742,0	.	.	.	.	.	.	.	.	.

jydu commented:
Hi,

Can you please also provide the corresponding command line? At first sight,
it looks like your file has a lot of missing genotypes, which might explain
why Fst cannot be calculated.

Julien.

On Mon, Sep 10, 2018 at 4:06 PM cfyin <notifications@github.com> wrote:

> ##fileformat=VCFv4.2
> ...
> ##samtoolsVersion=0.1.19-44428cd
> #CHROM POS ID REF ALT QUAL FILTER INFO FORMAT variant variant10 variant11
> variant12 variant13 variant14 variant15 variant2 variant3 variant4 variant5
> variant6 variant7 variant8 variant9
> 8 35518 . G C 753 .
> AC=4;AF=1.00;AN=4;DP=147;MLEAC=1;MLEAF=1.00;MQ=60.00;MQ0=0;set=variant-variant2-variant6-variant13
> GT:AD:DP:GQ:PL 1:0,32:32:99:783,0 . . . 1:0,28:28:99:704,0 . .
> 1:1,34:35:99:840,0 . . . 1:2,50:52:99:1245,0 . . .
> 8 35625 . C T 720 .
> AC=4;AF=1.00;AN=4;DP=158;MLEAC=1;MLEAF=1.00;MQ0=0;set=variant-variant2-variant6-variant13
> GT:AD:DP:GQ:PL 1:3,36:39:99:750,0 . . . 1:0,26:26:99:599,0 . .
> 1:0,47:47:99:1113,0 . . . 1:2,44:46:99:1041,0 . . .
> 9 64 . G C 6596 .
> AC=2;AF=1.00;AN=2;DP=591;MLEAC=1;MLEAF=1.00;MQ=60.00;MQ0=0;set=variant6-variant7
> GT:AD:DP:GQ:PL . . . . . . . . . . . 1:10,280:290:99:6626,0
> 1:2,292:294:99:6946,0 . .
> 9 85 . A T 2143 .
> AC=2;AF=1.00;AN=2;DP=115;FS=0.000;MLEAC=1;MLEAF=1.00;MQ=60.00;MQ0=0;set=variant10-variant14
> GT:AD:DP:GQ:PL . 1:0,55:55:99:2173,0 . . . 1:0,60:60:99:2318,0 . . . . . .
> . . .
> 9 203 . T G 2615 .
> AC=1;AF=1.00;AN=1;DP=57;FS=0.000;MLEAC=1;MLEAF=1.00;MQ=60.00;MQ0=0;QD=27.35;SOR=0.960;set=variant14
> GT:AD:DP:GQ:PL . . . . . 1:0,57:57:99:2645,0 . . . . . . . . .
> 9 2905 . A C 2618 .
> AC=1;AF=1.00;AN=1;DP=57;FS=0.000;MLEAC=1;MLEAF=1.00;MQ=60.00;MQ0=0;QD=27.60;SOR=0.877;set=variant14
> GT:AD:DP:GQ:PL . . . . . 1:0,57:57:99:2648,0 . . . . . . . . .
> 9 2906 . G T 2611 .
> AC=1;AF=1.00;AN=1;DP=57;FS=0.000;MLEAC=1;MLEAF=1.00;MQ=60.00;MQ0=0;QD=35.09;SOR=0.877;set=variant14
> GT:AD:DP:GQ:PL . . . . . 1:0,57:57:99:2641,0 . . . . . . . . .
> 9 2913 . A G 2608 .
> AC=1;AF=1.00;AN=1;DP=58;FS=0.000;MLEAC=1;MLEAF=1.00;MQ=60.00;MQ0=0;QD=33.34;SOR=0.836;set=variant14
> GT:AD:DP:GQ:PL . . . . . 1:0,58:58:99:2638,0 . . . . . . . . .
> 9 2704 . G T 2491 .
> AC=2;AF=1.00;AN=2;DP=128;FS=0.000;MLEAC=1;MLEAF=1.00;MQ=60.00;MQ0=0;set=variant10-variant14
> GT:AD:DP:GQ:PL . 1:0,62:62:99:2521,0 . . . 1:0,66:66:99:2681,0 . . . . . .
> . . .
> 9 2773 . T C 2627 .
> AC=2;AF=1.00;AN=2;DP=133;FS=0.000;MLEAC=1;MLEAF=1.00;MQ=60.00;MQ0=0;set=variant10-variant14
> GT:AD:DP:GQ:PL . 1:0,67:67:99:2657,0 . . . 1:0,66:66:99:2742,0 . . . . . .
> . . .
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/vcftools/vcftools/pull/69#issuecomment-419924871>, or mute
> the thread
> <https://github.com/notifications/unsubscribe-auth/ALLFJdjryLA7jgex2kjNwK9p028IwYxvks5uZnHcgaJpZM4LqXZ7>
> .
>


-- 
Julien Y. Dutheil, Ph-D
0 (+49) 4522 763 298

§ Max Planck Institute for Evolutionary Biology
Molecular Systems Evolution
Department of Evolutionary Genetics
Plön -- GERMANY

§ Institute of Evolutionary Sciences - Montpellier
University of Montpellier 2 -- FRANCE

jydu commented:
Hi,

You need to specify at least two population to compute Fst, for instance
vcftools --vcf mysnps.vcf --haploid --weir-fst-pop pop1 --weir-fst-pop pop2
where pop1 and pop2 are 2 files listing the genotypes for each population.
Is that what you are doing?
Concerning the missing data, you could try first to restrict your data to a
combination of SNPs and individuals without missing data, to see if the
issue persists. I am not exactly sure how the Fst calculation handles
missing data.

On Tue, Sep 11, 2018 at 8:23 AM cfyin <notifications@github.com> wrote:

> Hi,
> I only use three parameters for vcftools: --vcf, --weir-fst-pop, and
> --haploid.
> The missing genotypes (the points int colcumns >= 10) leads to Fst error?
> I used the vcf-merge or GATK-CombineVariants to combine the VCF files from
> each samples, by default the "missing genotype" is the same as the
> reference genome.
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/vcftools/vcftools/pull/69#issuecomment-420160847>, or mute
> the thread
> <https://github.com/notifications/unsubscribe-auth/ALLFJbF-68J44CnXSvPJGyJc6uDDu7Rxks5uZ1b7gaJpZM4LqXZ7>
> .
>


-- 
Julien Y. Dutheil, Ph-D
0 (+49) 4522 763 298

§ Max Planck Institute for Evolutionary Biology
Molecular Systems Evolution
Department of Evolutionary Genetics
Plön -- GERMANY

§ Institute of Evolutionary Sciences - Montpellier
University of Montpellier 2 -- FRANCE

cfyin commented:
Yes, your example is my command line.
I replaced the GT (genotype) of the missing data with 0 as descripted in VCF 4.2 fomat, 
but the results remained 'nan'.
I replaced the GT (genotype) of the missing data with 0/0 and the others with 1/1, 
and use the default diploid model (no use the --haploid), the results is normal.

jydu commented:
Hi again,

would you mind sending me your pop1 and pop2 files then so that I try to
reproduce the error and hopefully fix it?

Best,

Julien.

On Tue, Sep 11, 2018 at 3:04 PM cfyin <notifications@github.com> wrote:

> Yes, your example is my command line.
> I replaced the GT (genotype) of the missing data with 0 as descripted in
> VCF 4.2 fomat,
> but the results remained 'nan'.
> I replaced the GT (genotype) of the missing data with 0/0 and the others
> with 1/1,
> and use the default diploid model (no use the --haploid), the results is
> normal.
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/vcftools/vcftools/pull/69#issuecomment-420266830>, or mute
> the thread
> <https://github.com/notifications/unsubscribe-auth/ALLFJTmIIA_z1T5JzSqGpaTVtf-iHgRsks5uZ7TQgaJpZM4LqXZ7>
> .
>


-- 
Julien Y. Dutheil, Ph-D
0 (+49) 4522 763 298

§ Max Planck Institute for Evolutionary Biology
Molecular Systems Evolution
Department of Evolutionary Genetics
Plön -- GERMANY

§ Institute of Evolutionary Sciences - Montpellier
University of Montpellier 2 -- FRANCE

Commit:
Fixed Fst calculation for haploids.

jydu commented:
Hi,

It should be now fixed (on jydu/vcftools). Please check that the results are as expected... as vcftools is intrinsically built around diploid SNPs, my approach is to count haploid SNPs as homozygous, after discarding any diploid SNP that might be present. So this is like duplicating all SNPs, counting them twice... This should give the correct results I think, but again, I cannot exclude that there is something I've not thought about!

jydu commented:
Negative Fst values are typically interpreted as 0. As for the missing data, I would be very careful about replacing them with 0, the reference allele, you might be creating some bias there! If you think of a theoretical scenario where two populations have fixed the derived allele 1, but in one population that particular position was not called, then you're going to set it to 0. You will then count this SNP as perfectly differentiated between the two pops when it's not. Not sure how vcftools handles missing data for Fst calculations, but maybe there exists some implementation that properly accounts for them.

Niohuruzh commented:
Hi jydu,
I `configure & make & make install` the jydu/vcftools. And vcftools version is 0.1.17. But I cannot use haploid mode. It shows that:
`Error: Unknown option; --haploid`

Do you have any idea to solve it?
Best regards!

jydu commented:
Hi,

I tested and I cannot reproduce the issue:
```
./src/cpp/vcftools --vcf examples/valid-4.1.vcf --window-pi 10

VCFtools - 0.1.17
(C) Adam Auton and Anthony Marcketta 2009

Parameters as interpreted:
	--vcf examples/valid-4.1.vcf
	--window-pi 10

Warning: Expected at least 2 parts in INFO entry: ID=DB,Number=0,Type=Flag,Description="dbSNP membership, build 129">
After filtering, kept 3 out of 3 Individuals
Outputting Windowed Nucleotide Diversity Statistics...
	windowPi: Only using fully diploid sites.
After filtering, kept 11 out of a possible 12 Sites
```

vs.

```
./src/cpp/vcftools --vcf examples/valid-4.1.vcf --haploid --window-pi 10

VCFtools - 0.1.17
(C) Adam Auton and Anthony Marcketta 2009

Parameters as interpreted:
	--vcf examples/valid-4.1.vcf
	--window-pi 10

Warning: Expected at least 2 parts in INFO entry: ID=DB,Number=0,Type=Flag,Description="dbSNP membership, build 129">
After filtering, kept 3 out of 3 Individuals
Outputting Windowed Nucleotide Diversity Statistics...
	windowPi: Only using fully haploid sites.
After filtering, kept 1 out of a possible 12 Sites
```

Which command do you use?

Niohuruzh commented:
Hi jydu,
Thanks for your reply.
I use `vcftools-0.1.17` with the following command:
`vcftools --vcf snp.vcf --haploid --weir-fst-pop 1_population.txt --weir-fst-pop 2_population.txt`
and then, the result back:
`Error: Unknown option: --haploid`

BTW type `vcftools --version`
back `VCFtools (0.1.17)`

Looking forward your reply
Best regards!

jydu commented:
I'm really not sure what is going on, it really looks like the wrong version is used. Assuming that you cloned from https://github.com/jydu/vcftools.git and not from this repository, could it be that multiple versions are installed on your system? Could you try to run the commands I pasted before in the vcftools source directory? You should get the same result as I had, otherwise the wrong version is installed.

Niohuruzh commented:
Hi jydu,
Thanks for your help. I have solved my problem. The key is to `git clone https://github.com/jydu/vcftools.git`
And my vcftools `--haploid` results is approximately same with R(popgenome) and R(heirfstat)
But I want to know why official vcftools software`(version=0.1.16)` don't use hapoloid mode? Because the results from haploid mode is not true?

Looking forward your reply
Best regards!

jydu commented:
Originally, vcftools discards haploid SNPs. I am not a developper of the vcftools, I modified the code to add the haploid option for my own usage and shared it. Whether this option should become part of the official vcftools package is up to its developpers :)
Note that the modification does not alter the calculations, which are the same as for dipoids. In haploid mode, haploid SNPs are kept and diploid SNPs are ignored.

Best,

mmokrejs commented:
@Niohuruzh vcftools are not maintained anymore for a long while, look at the bcftools which is the successor tool. All patches and feature additions should be sent to the bcftools tree, as welll as results of tests. Please note I am not a [bv]cftools developer.

jydu commented:
Hi,

Not an easy one... If the diploid samples are phased, I would convert them to haploids, and treat the whole dataset as haploid. If they are unphased, this is more tricky I think and I do not have a solution out-of-the-box...

Best,

Julien.

jydu commented:
hi,

I would just separate the two alleles into separate haploid individuals, like indv1.1 and indv1.2, etc. By the way, if you only need Fst or diversity-based statistics, you actually do not even need phasing, as these statistics do not consider linkage. I'm not aware of a program that would do the conversion, but it should be possible to write one with packages such as vcfR for instance.
As for the --haploid option, it should work with windowed-Fst.

Hope this helps,

Julien.

Honeyman88 commented:
Thanks a lot. I'll have a look at vcfR. For the --haploid option, I just tested the converted data I made. But I got error that said the --haploid is an unknown option?
![image](https://user-images.githubusercontent.com/93487645/163177216-f0855259-5cd2-4e97-baec-65e4ebd94bc6.png)

jydu commented:
I think it is the usual error... you need to use the version from jydu/vcftools, not the official vcftools/vcftools, because the merge request has not been accepted. So the --haploid tag is not part of the official vcftools release.

Honeyman88 commented:
Thanks. Ah yes, I should not have copied directly from here, 
![image](https://user-images.githubusercontent.com/93487645/163219092-ce26d486-56cf-4735-be53-8ba8ef177145.png)
Instead I should use this **git clone https://github.com/jydu/vcftools.git**. Now it works. Appreciated a lot!
Cheers

snackens commented:
Hi, thank you for your excellent tools.

I tried to calculate Fst value using --haploid mode, but the values were slightly different from values calculated by popgenome R package. While some values were NA in Popgenome, Fst values of the SNP sites were not NA in vcftools.
If you know, could you please teach me the reasons?

Thanks,

AdrianAllen1977 commented:
Hi, I have bacterial sequence data that I'd like to run some Tajima's D analyses for.  The vcf files I have are called via a SNP pipeline explicitly for bacterial genomics.  

I can merge multiple vcfs using bcftools and use the -0 function to set all missing sites as the reference call and then run the original version of vcftools Tajima D analysis - getting an output.  When I try to use your fork version with the --haploid switch, I'm getting an output but one with just the column headers.

Is this something to do with the original format of my vcf files?  I've attached one here.
[BTB_Gen_001_Bov_2001_1.140_LT708304_q30.vcf.gz](https://github.com/vcftools/vcftools/files/9954074/BTB_Gen_001_Bov_2001_1.140_LT708304_q30.vcf.gz)

jydu commented:
Hi,

As far as I can see, your SNPs are encoded as unphased diploid. Using --haploid will ignore all diploid SNPs, and, therefore, there is nothing left!

Julien.

AdrianAllen1977 commented:
Thanks Julien, I suspected it was down to the way the vcf was coded. If I
change the genotype calls from 0/0 to 0 and 1/1 to 1, the haploid script
should then work I imagine.

Is it likely to be very different from the result I'm getting just running
the original vcftools Tajima script??

I'll take a look tomorrow. Thanks for.the quick reply and for making such
useful tools.

Best regards

Adrian

On Mon, 7 Nov 2022, 17:57 Julien Y. Dutheil, ***@***.***>
wrote:

> Hi,
>
> As far as I can see, your SNPs are encoded as unphased diploid. Using
> --haploid will ignore all diploid SNPs, and, therefore, there is nothing
> left!
>
> Julien.
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/vcftools/vcftools/pull/69#issuecomment-1305981339>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ALOE2I5XKSK37BQNO7UD7MTWHE7HBANCNFSM4C5JOZ5Q>
> .
> You are receiving this because you commented.Message ID:
> ***@***.***>
>

jydu commented:
Hi Adrian,

I would expect the results to be different. As a back-of-the-envelope calculation, I would say that the number of segregating positions is not affected, so that Watterson's theta is "only" erroneous by the sample size, which is off by a factor of two. However, all individuals are pure homozygotes, so that the average pairwise heterozygosity should be highly underestimated.

Best,

Julien.

AdrianAllen1977 commented:
Hi Julien, yes that makes perfect sense and is indeed the case having run
the analysis now.

Thanks for the help and the haploid tools. Will put them to good use.

Is there a preferred way to cite you on this? Maybe just cite the github
fork?

Thanks again and best regards

Adrian

On Wed, 9 Nov 2022, 05:37 Julien Y. Dutheil, ***@***.***>
wrote:

> Hi Adrian,
>
> I would expect the results to be different. As a back-of-the-envelope
> calculation, I would say that the number of segregating positions is not
> affected, so that Watterson's theta is "only" erroneous by the sample size,
> which is off by a factor of two. However, all individuals are pure
> homozygotes, so that the average pairwise heterozygosity should be highly
> underestimated.
>
> Best,
>
> Julien.
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/vcftools/vcftools/pull/69#issuecomment-1308237261>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ALOE2I3LZUTVSQSBIFPBTPLWHM2BLANCNFSM4C5JOZ5Q>
> .
> You are receiving this because you commented.Message ID:
> ***@***.***>
>

jydu commented:
Good question... I would simply use the publication of the official VCFTools, and mention the fork in the mat & met, something like that. The modification for the haploid mode is really nothing: just discarding diploid SNPs and keep the haploid ones, and scale diversity estimators by 2 instead of 4.

jydu commented:
Please install this version: https://github.com/jydu/vcftools/
The pull request has so far not been accepted into the main vcftools.

Julien.

jydu committed:
Merge branch 'vcftools:master' into master

Commit:
Formatting.

alephreish commented:
@auton1 Any chances of merging this PR?

jydu commented:
Hi, as this is not part of the official vcftools, I think you need to mention the fork in the material and methods. Something like "... using a modified version of the vcftools [insert reference to original vcftools paper] to work on haploid genotypes (available at https://github.com/jydu/vcftools)".

JaSalg commented:
Hello, 
I am sorry if this is too much of  a basic question, but I have tried looking around and have not found a solution. 

I am interested in installing your branch of vcftools for the --halpoid mode.

I had vcftools previously in my ubuntu VM so I erased all vcftools data, then ran:

git clone https://github.com/jydu/vcftools.git
cd vcftools
./autogen.sh
./configure
make
make install

But when then calling vcftools, I would still have the 0.1.16 version, which is the master branch IIRC, instead of your version which I believe is 0.1.17. 

Any tips to solve this? 

Best regards

<img width="761" height="379" alt="vcftools1" src="https://github.com/user-attachments/assets/cf38174c-9501-47bf-bc01-bee2a3a4fe71" />
<img width="609" height="350" alt="vcftools2" src="https://github.com/user-attachments/assets/b1eb9426-1d4d-4820-ad4a-877c2c4615b7" />

jydu commented:
Hi, indeed! But your compilation did not work. It says that autoreconf is not found :s On ubuntu, you need to install the autoconf package for that:
```
sudo apt install autoconf
```
Does that help?

Julien.

JaSalg commented:
Hello Julien, 

First of all thank you very much for the answer, as a molecular biologist I am very lost with Linux console. 

I did the

> sudo apt install autoconf

And now it does run the ./autogen and ./configure but there seems to be an issue during autoconfigure, and it cannot go to the Make or Make Install steps. 

<img width="806" height="528" alt="VCF troubleshooting3" src="https://github.com/user-attachments/assets/f00c12aa-f2ac-45a4-a3d7-b87d3f310abe" />

Best regards, 

Jaume

jydu commented:
Dear Jaume,

Could it be that you do not have zlib installed?
Does
```
sudo apt install zlib1g-dev
```
help by any chance?

jydu committed:
Merge branch 'vcftools:master' into master

JaSalg commented:
Hello, 

I am quite sure I already had zlib installed, I still reinstalled it, but running the commands I said before I get the exact same error messages, so no improvement unfortunately

JaSalg commented:
Hello, 

I write to say that I have solved it. 

Not by my own hand, a friend who is a senior programmer in non-bio related stuff helped. 

Rather than only installing Zlib i had to do this

`sudo apt install autoconf automake libtool pkg-config zlib1g-dev`

And it does seem that even if you run the `./autogen.sh` it does not properly execute. 

So I typed manually `autoreconf -fi` before the rest. 

Also, it does not work if you do the commands one at a time, you have to do all of them at the same time for it to properly install, at least in my machine. 

So in the end it was (with everything installed properly)

```
git clone https://github.com/jydu/vcftools.git
cd vcftools
./autogen.sh
autoreconf -fi
./configure
make
sudo make install
```

Now I have the 0.1.18 version, and --haploid is detected as a command!

jydu commented:
Good that it worked out!
